lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1463,56 +1463,78 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
         Value rankVal = rewriter.create<Torch::ConstantIntOp>(
             binder.getLoc(), rewriter.getType<Torch::IntType>(),
             rewriter.getIntegerAttr(rewriter.getIntegerType(64), rank));
-        Value zero = rewriter.create<Torch::ConstantIntOp>(
-            loc, rewriter.getI64IntegerAttr(0));
+        Value cstZero = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(0));
+        Value cstOne = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(1));
 
         Value axisScalar = rewriter.create<Torch::AtenItemOp>(
             binder.getLoc(), rewriter.getType<Torch::IntType>(), axisTensor);
         Value isNegative = rewriter.create<Torch::AtenLtIntOp>(
-            binder.getLoc(), axisScalar, zero);
+            binder.getLoc(), axisScalar, cstZero);
         isNegative =
             rewriter.create<Torch::AtenIntBoolOp>(binder.getLoc(), isNegative);
         Value finalOffset = rewriter.create<Torch::AtenMulIntOp>(
             binder.getLoc(), isNegative, rankVal);
-        Value dim = rewriter.create<Torch::AtenAddIntOp>(
+        Value axis = rewriter.create<Torch::AtenAddIntOp>(
             binder.getLoc(), axisScalar, finalOffset);
+        Value none = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
 
-        Torch::BaseTensorType resultTensorType =
-            cast<Torch::BaseTensorType>(resultType);
-        if (!resultTensorType.hasDtype()) {
-          return rewriter.notifyMatchFailure(
-              binder.op, "expected result type to have a dtype");
+        if (!reverse) {
+          if (!exclusive) {
+            rewriter.replaceOpWithNewOp<Torch::AtenCumsumOp>(
+                binder.op, resultType, operand, axis, none);
+            return success();
+          } else {
+            Value cumsum = rewriter.create<Torch::AtenCumsumOp>(
+                binder.getLoc(), resultType, operand, axis, none);
+            rewriter.replaceOpWithNewOp<Torch::AtenSubTensorOp>(
+                binder.op, resultType, cumsum, operand, cstOne);
+            return success();
+          }
         }
-        // resultTensorType.print(llvm::outs());
-        Value none = rewriter.create<Torch::ConstantNoneOp>(loc);
-        rewriter.replaceOpWithNewOp<Torch::AtenCumsumOp>(binder.op, resultType,
-                                                         operand, dim, none);
+
+        Value dims = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            rewriter.getType<Torch::ListType>(
+                rewriter.getType<Torch::IntType>()),
+            SmallVector<Value>{axis});
+        Value flip = rewriter.create<Torch::AtenFlipOp>(
+            binder.getLoc(), resultType, operand, dims);
+        Value cumsum = rewriter.create<Torch::AtenCumsumOp>(
+            binder.getLoc(), resultType, flip, axis, none);
```

Comment:
To simplify this entire flow, you can do:
```
if (reverse)
    operand = AtenFlipOp(operand)
cumsum = AtenCumSumOp(operand)
if (!exclusive)
    return cumsum;
cumsum = cumsum - operand;
return cumsum;
```

Reply:
It needs to sub the original `operand` if exclusive, `operand = AtenFlipOp(operand)` will change the operand.

Reply:
I simplify it by using `replaceOp`.